src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -86,55 +85,44 @@ public class SendReceiveTest : ClientWebSocketTestBase
             {
-                var cts = new CancellationTokenSource(TimeOutMilliseconds);
+                var tasks = new Task[2];
 
-                Task[] tasks = new Task[10];
+                // Need large buffer, or Send completes too quickly and no exception is thrown.
+                var sendBuffer = new byte[128 * 1024];
+                var segment = new ArraySegment<byte>(sendBuffer);
 
-                try
+                for (int i = 0; i < tasks.Length; i++)
                 {
-                    for (int i = 0; i < tasks.Length; i++)
-                    {
-                        tasks[i] = cws.SendAsync(
-                            WebSocketData.GetBufferFromText("hello"),
-                            WebSocketMessageType.Text,
-                            true,
-                            cts.Token);
-                    }
+                    tasks[i] = cws.SendAsync(
+                        segment,
+                        WebSocketMessageType.Text,
+                        true,
+                        CancellationToken.None);
+                }
 
-                    Task.WaitAll(tasks);
+                await Assert.ThrowsAsync<InvalidOperationException>(() => Task.WhenAll(tasks));
+                Assert.Equal(WebSocketState.Aborted, cws.State);
 
-                    Assert.Equal(WebSocketState.Open, cws.State);
-                }
-                catch (AggregateException ag)
+                Assert.All(tasks, task =>
                 {
-                    foreach (var ex in ag.InnerExceptions)
-                    {
-                        if (ex is InvalidOperationException)
-                        {
-                            Assert.Equal(
-                                ResourceHelper.GetExceptionMessage(
-                                    "net_Websockets_AlreadyOneOutstandingOperation",
-                                    "SendAsync"),
-                                ex.Message);
-
-                            Assert.Equal(WebSocketState.Aborted, cws.State);
-                        }
-                        else if (ex is WebSocketException)
-                        {
-                            // Multiple cases.
-                            Assert.Equal(WebSocketState.Aborted, cws.State);
-
-                            WebSocketError errCode = (ex as WebSocketException).WebSocketErrorCode;
-                            Assert.True(
-                                (errCode == WebSocketError.InvalidState) || (errCode == WebSocketError.Success),
-                                "WebSocketErrorCode");
-                        }
-                        else
-                        {
-                            Assert.IsAssignableFrom<OperationCanceledException>(ex);
-                        }
-                    }
-                }
+                    Assert.Equal(TaskStatus.Faulted, task.Status);
```

Comment:
Argh, except the server is still picking up and buffering the connection, even if the server doesn't actually process it yet.
IOW:

1. send: delay 5 seconds - completes instantly
2. send: some text - completes instantly
3. receive: some text - completes after 5 seconds

...whereas I need step 2 to complete after 5 seconds.  Reading through the server code, it looks like the actual pickup/process on their side should be happening after 5 seconds, at least from that layer's view.

@stephentoub - relying on the size of the message sent isn't likely to work long term either (as connections speeds will increase over time).  Increasing the number helps the chances of a conflict, but still isn't a guarantee; I'd prefer it if there were some way to tell the server to delay pickup (which would remove most of the possibility for race conditions), but that looks like it would require writing the echo server on a lower layer.  Thoughts?  I haven't investigated the whole stack, so it's possible the send task is returned as "completed" even if it hasn't left the box yet, but I'm figuring this is unlikely.